subcommands/devices/cmd.go

```diff
@@ -61,3 +62,14 @@ func NewCommand() *cobra.Command {
 	cmd.AddCommand(updatesCmd)
 	return cmd
 }
+
+func getDeviceApi(_ *cobra.Command, name string) client.DeviceApi {
+	return api.DeviceApiByName(viper.GetString("factory"), name)
+}
+
+func getDevice(cmd *cobra.Command, name string) *client.Device {
+	dapi := getDeviceApi(cmd, name)
+	d, err := dapi.Get()
+	subcommands.DieNotNil(err)
+	return d
+}
```

Comment:
It seems to me that this change may be higher in the ~food chain~ commit list.
Then, the preceeding commits could gradually turn commands to the final desired shape (presented here), without an intermediate changes (and then one big commit) like it is now.

My gut feeling is that it would be easier to navigate this PR (and cause less code churn).
WDYT?